datafusion/src/physical_plan/hash_join.rs

```diff
@@ -476,18 +483,14 @@ fn update_hash(
 
     // insert hashes to key of the hashmap
     for (row, hash_value) in hash_values.iter().enumerate() {
-        match hash.raw_entry_mut().from_hash(*hash_value, |_| true) {
-            hashbrown::hash_map::RawEntryMut::Occupied(mut entry) => {
-                entry.get_mut().push((row + offset) as u64);
-            }
-            hashbrown::hash_map::RawEntryMut::Vacant(entry) => {
-                entry.insert_hashed_nocheck(
-                    *hash_value,
-                    (),
-                    smallvec![(row + offset) as u64],
-                );
-            }
-        };
+        let item = hash_map.0.get_mut(*hash_value, |_| true);
```

Comment:
I'm not so sure this is correct anymore. I think it should be the same as in the previous implementation, but I wonder whether the hash values still need to be checked, as also hash values might end up in the same bucket. Maybe it's easier to test this now with the raw table api.

Reply:
I reverted this to check on equality of hashes.
I found it impossible to test this, as creating a hash which points to the same bucket is surprisingly hard (tried 100M values) but might happen more frequently maybe when there is a lot of data in the hashmap.